src/main/java/de/tum/in/www1/artemis/service/ProgrammingSubmissionService.java

```diff
@@ -634,4 +633,42 @@ public ProgrammingSubmission findByIdWithEagerResultAndFeedback(long submissionI
         return programmingSubmissionRepository.findWithEagerResultAssessorFeedbackById(submissionId)
                 .orElseThrow(() -> new EntityNotFoundException("Programming submission with id \"" + submissionId + "\" does not exist"));
     }
+
+    /**
+     * Get a programming submission of the given exercise that still needs to be assessed and lock the submission to prevent other tutors from receiving and assessing it.
+     *
+     * @param exercise the exercise the submission should belong to
+     * @return a locked programming submission that needs an assessment
+     */
+    @Transactional
```

Comment:
please remove this `@Transactional` here

Reply:
Isn't it possible then, that two tutors aquire the same lock on a submission?

Reply:
Done